nion/swift/ExportDialog.py

```diff
@@ -34,167 +36,191 @@
+            self.ui.set_persistent_string("export_directory", self.directory)
+
+    def on_writer_changed(self, widget: Declarative.UIWidget, current_index: int) -> None:
+        writer = self.writers[current_index]
+        self.viewmodel.writer.value = writer
+
+    def _build_ui(self, u: Declarative.DeclarativeUI) -> None:
+        self.file_type_index = 0
+        self.writers = ImportExportManager.ImportExportManager().get_writers()
+        writers_names = [getattr(writer, "name") for writer in self.writers]
+        self.file_types = writers_names
+
+        #  self.viewmodel = viewmodel
+
+        # Export Folder
+        directory_label = u.create_row(u.create_label(text="Location:"))
+        directory_text = u.create_row(u.create_label(text=f'@binding(viewmodel.directory.value)'))
+        self.directory_text_label = directory_text
+        directory_button = u.create_row(u.create_push_button(text="Select Path..", on_clicked='choose_directory'))
+
+        # Filename
+        filename_label = u.create_row(u.create_label(text="Filename:"))
+        ## Title
+        title_checkbox = u.create_row(u.create_check_box(text="Include Title", checked=f'@binding(viewmodel.include_title.value)'))
+
+        ## Date
+        date_checkbox = u.create_row(u.create_check_box(text="Include Date", checked=f'@binding(viewmodel.include_date.value)'))
+
+        ## Dimensions
+        dimension_checkbox = u.create_row(u.create_check_box(text="Include Dimensions", checked=f'@binding(viewmodel.include_dimensions.value)'))
+
+        ## Sequence Number
+        sequence_checkbox = u.create_row(u.create_check_box(text="Include Sequence Number", checked=f'@binding(viewmodel.include_sequence.value)'))
+
+        ## Prefix
+        prefix_checkbox = u.create_row(u.create_check_box(text="Include Prefix", checked=f'@binding(viewmodel.include_prefix.value)'))
+        prefix_textbox = u.create_row(u.create_text_edit(text=f'@binding(viewmodel.prefix.value)'))
+
+        ## File Type
+        file_type_combobox = u.create_combo_box(items=self.file_types, current_index=f'@binding(file_type_index.value)', on_current_index_changed="on_writer_changed")
+
+        ## Build final ui column
+        column = u.create_column(directory_label,
+                                 directory_text,
+                                 directory_button,
+                                 filename_label,
+                                 title_checkbox,
+                                 date_checkbox,
+                                 dimension_checkbox,
+                                 sequence_checkbox,
+                                 prefix_checkbox,
+                                 prefix_textbox,
+                                 file_type_combobox,
+                                 spacing=12, margin=12)
+        self.ui_view = column
+
+    @staticmethod
+    def build_filename(components: typing.List[str], extension: str, suffix: str = "($)", path: str = "") -> str:
```

Comment:
The `suffix` argument is unused. This looks partially implemented. So... when splitting into the two PR's, you can probably just drop it. Then let's discuss the unique filename argument. I'd rather see a slightly more holistic approach where we keep track of what filenames have been used and only then make the subsequent matching filenames unique. I think it's difficult to do it with a single `build_filename` method unless a `dict` of used base filenames and their count is passed in.

Reply:
We can discuss here; but at this point, it's unused code, so it can be removed from this PR, right?

Reply:
Ok so the suffix part has been removed as you are correct, it is unused.  Onto the actual unique filename bit.

So the requirement as I understood it was to make sure that when exporting files to not automatically overwrite so the user did not lose data, especially in the case where the parameters selected meant that multiple exported images would have the same filename.  The simplest solution to this is to follow the Windows paradigm where if attempting to create a file with the same name to simply add an incrementing suffix in brackets after the filename.

You have mentioned some dictionary of data items to filenames, but I have some questions about that.  Does it actually provide any extra benefit over the standard Windows method? How should it handle already existing files in that directory that are not in that dictionary?